clients/drcachesim/drpt2trace/drpt2trace.cpp

```diff
@@ -294,21 +300,26 @@ main(int argc, const char *argv[])
     config.sb_config.tsc_offset = op_sb_tsc_offset.get_value();
     config.sb_config.kernel_start = op_sb_kernel_start.get_value();
 
-    /* Convert the pt raw data to DR IR. */
+    /* Convert the PT raw trace to DR IR. */
     std::unique_ptr<pt2ir_t> ptconverter(new pt2ir_t());
     if (!ptconverter->init(config)) {
         std::cerr << CLIENT_NAME << ": failed to initialize pt2ir_t." << std::endl;
         return FAILURE;
     }
-    if (ptconverter->convert() != PT2IR_CONV_SUCCESS) {
-        std::cerr << CLIENT_NAME << ": failed to convert pt raw data to DR IR."
+    instrlist_t *ilist = nullptr;
+    pt2ir_convert_status_t status = ptconverter->convert(&ilist);
+    if (status != PT2IR_CONV_SUCCESS) {
+        std::cerr << CLIENT_NAME << ": failed to convert PT raw trace to DR IR."
                   << std::endl;
+        instrlist_clear_and_destroy(GLOBAL_DCONTEXT, ilist);
```

Comment:
In case of failure, it's better if the convert routine itself destroys the ilist.

Reply:
Done.